circuits/circom/core/qv/processMessages.circom

```diff
@@ -0,0 +1,562 @@
+    // The coordinator's private key.
+    signal input coordPrivKey;
+    // The cooordinator's public key (derived from the contract).
+    signal input coordPubKey[2];
+    // The ECDH public key per message.
+    signal input encPubKeys[batchSize][2];
+    // The current state root (before the processing).
+    signal input currentStateRoot;
+
+    // The state leaves upon which messages are applied.
+    //    transform(currentStateLeaf[4], message5) => newStateLeaf4
+    //    transform(currentStateLeaf[3], message4) => newStateLeaf3
+    //    transform(currentStateLeaf[2], message3) => newStateLeaf2
+    //    transform(currentStateLeaf[1], message1) => newStateLeaf1
+    //    transform(currentStateLeaf[0], message0) => newStateLeaf0
+    //    ...
+
+    signal input currentStateLeaves[batchSize][STATE_LEAF_LENGTH];
+    // The Merkle path to each incremental new state root.
+    signal input currentStateLeavesPathElements[batchSize][stateTreeDepth][TREE_ARITY - 1];
+    // The salted commitment to the state and ballot roots.
+    signal input currentSbCommitment;
+    signal input currentSbSalt;
+    // The salted commitment to the new state and ballot roots.
+    signal input newSbCommitment;
+    signal input newSbSalt;
+    // The ballots before message processing.
+    signal input currentBallotRoot;
+    // Intermediate ballots.
+    signal input currentBallots[batchSize][BALLOT_LENGTH];
+    signal input currentBallotsPathElements[batchSize][stateTreeDepth][TREE_ARITY - 1];
+    // Intermediate vote weights.
+    signal input currentVoteWeights[batchSize];
+    signal input currentVoteWeightsPathElements[batchSize][voteOptionTreeDepth][TREE_ARITY - 1];
+
+    // nb. The messages are processed in REVERSE order.
+    // Therefore, the index of the first message to process does not match the index of the
+    // first message (e.g., [msg1, msg2, msg3] => first message to process has index 3).
+
+    // The index of the first message leaf in the batch, inclusive.
+    signal batchStartIndex;
+    
+    // The index of the last message leaf in the batch to process, exclusive.
+    // This value may be less than batchStartIndex + batchSize if this batch is
+    // the last batch and the total number of mesages is not a multiple of the batch size.
+    signal batchEndIndex;
+
+    // The history of state and ballot roots and temporary intermediate
+    // signals (for processing purposes).
+    signal stateRoots[batchSize + 1];
+    signal ballotRoots[batchSize + 1];
+    signal tmpStateRoot1[batchSize];
+    signal tmpStateRoot2[batchSize];
+    signal tmpBallotRoot1[batchSize];
+    signal tmpBallotRoot2[batchSize];
+
+    // Must verify the current sb commitment.
+    var currentSbCommitmentHash = PoseidonHasher(3)([currentStateRoot, currentBallotRoot, currentSbSalt]);
+    currentSbCommitmentHash === currentSbCommitment;
+
+    // Verify "public" inputs and assign unpacked values.
```

Comment:
I know this was here already, but wondering why public is in quote 🤔

Reply:
This is a great question - let me remove the double quotes.